src/code/sourceLocationTracker.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+var EventManager = require('../lib/eventManager')
+var util = require('../helpers/global')
+var helper = require('../helpers/traceHelper')
+var SourceMappingDecoder = require('../util/sourceMappingDecoder')
+
+/**
+ * Process the source code location for the current executing bytecode
+ */
+function SourceLocationTracker (_codeManager) {
+  this.codeManager = _codeManager
+  util.extend(this, new EventManager())
+  this.sourceMappingDecoder = new SourceMappingDecoder()
+}
+
+/**
+ * Return the source location associated with the given @arg index
+ *
+ * @param {String} address - object containing attributes {source} and {length}
+ * @param {Int} index - array returned by the function 'getLinebreakPositions'
+ * @param {Object} contractDetails - AST of compiled contracts
+ * @param {Function} cb - callback function
+ */
+SourceLocationTracker.prototype.getSourceLocation = function (address, index, contractsDetails, cb) {
```

Comment:
Instead of the AST of compiled contracts?
we still need to pass the address cause at this point we don't know the associated source map for the given context (address)